ui/packages/catalog-ui-search/src/main/webapp/component/query-src/query-src.view.js

```diff
@@ -16,12 +16,46 @@
 const Marionette = require('marionette')
 const _ = require('underscore')
 const $ = require('jquery')
-const template = require('./query-src.hbs')
 const CustomElements = require('js/CustomElements')
 const sources = require('component/singletons/sources-instance')
+import React from 'react'
 
 module.exports = Marionette.ItemView.extend({
-  template: template,
+  template(props) {
+    return (
+      <React.Fragment key={Math.random()}>
+        <div className="choice is-all is-available">
+          <span className="choice-text">All Sources</span>
+          <span className="choice-selected fa fa-check" />
+        </div>
+        {props.map(source => {
+          return (
+            <div
+              key={source.id}
+              className={`choice is-specific ${
+                source.available ? 'is-available' : ''
+              }`}
+              data-value={source.id}
+            >
+              <span className="choice-text">
+                {source.available ? (
+                  ''
+                ) : (
+                  <span className="fa fa-exclamation-triangle" />
+                )}
+                {source.local ? (
+                  <span className="fa source-icon fa-home" />
```

Comment:
✏️ 	Can this ternary be refactored to only wrap the `fa-home`/`fa-cloud` classes? I think it would help simplify this jsx. Something like
`<span className="fa source-icon {source.local ? "fa-home" : "fa-cloud"} />`